runhouse/servers/env_servlet.py

```diff
@@ -48,26 +41,21 @@ def wrapper(*args, **kwargs):
         # the exception if there is one, instead of returning a Response object.
         try:
             output = func(*args, **kwargs)
+            if serialization is None:
+                return output
             if output is not None:
-                if serialization is None:
-                    return output
-                else:
-                    serialized_data = serialize_data(output, serialization)
-                    return Response(
-                        output_type=OutputType.RESULT_SERIALIZED,
-                        data=serialized_data,
-                        serialization=serialization,
-                    )
+                serialized_data = serialize_data(output, serialization)
+                return Response(
+                    output_type=OutputType.RESULT_SERIALIZED,
+                    data=serialized_data,
+                    serialization=serialization,
+                )
             else:
                 return Response(
                     output_type=OutputType.SUCCESS,
                 )
         except Exception as e:
-            if serialization is None:
```

Comment:
why don't we want to `raise e` here directly?

Reply:
Inside `handle_exception_response` we have special handling for certain exceptions that we want to convert into HTTPExceptions (e.g. permission error -> 403), and barring that if serialization is None we just raise the exception as you're suggesting.